grails-app/services/com/unifina/service/UserAvatarImageService.groovy

```diff
@@ -0,0 +1,34 @@
+package com.unifina.service
+
+import com.unifina.api.ApiException
+import com.unifina.domain.security.SecUser
+import com.unifina.provider.FileUploadProvider
+import com.unifina.utils.IdGenerator
+import com.unifina.utils.ImageVerifier
+import grails.transaction.Transactional
+
+@Transactional
+class UserAvatarImageService {
+	private final long maxSize = 1024*1024*5
+	FileUploadProvider fileUploadProvider
+	ImageVerifier imageVerifier = new ImageVerifier(maxSize)
+	IdGenerator idGenerator = new IdGenerator()
+
+	def replaceImage(SecUser user, byte[] fileBytes, String filename) {
+		imageVerifier.verifyImage(fileBytes)
```

Comment:
That's right. Should it be used? @juhah @fonty1
I know there is an image resizer in the front end.

Reply:
The frontend can send the cropped image that is resized to fit the avatar editor or in the original size. But maybe good to add that to backend as well?

Reply:
@juhah Sure thing, what size in pixels for the final image?

Reply:
The "final" dimension in the avatar editor is 220px but it's never used as that big (mostly 80x80 and 40x40). But we could save it in that double size so it will look good in retina displays as well.

Reply:
These are such tiny images that it doesn't make sense to me to save both, that would be micro-optimization. @juhah is this what you meant (save both), or save 80x80 only? 

I'm offering a third option: These small images will use negligible bandwidth and storage anyway, so would it be more future-proof to save them in 220px? Maybe we later want to allow users to view a bigger version?

Reply:
I meant save only the 220px and then scale in browser. I mean that they are used only in those smaller "windows".